samcli/cli/main.py

```diff
@@ -117,6 +117,14 @@ def cli(ctx):
     import atexit
     from samcli.lib.telemetry.metric import send_installed_metric, emit_all_metrics
 
+    # if development version of SAM CLI is used, attach module proxy
+    # to catch missing configuration for dynamic/hidden imports
+    if ctx and getattr(ctx, "command_path", None) == "samdev":
```

Comment:
can we configure the binary name, and also make it such that its injectable (maybe env var?)? especially since we have multiple binary names such as samdev, sam-nightly and now sam-dev from your GH actions PR.

Reply:
As discussed offline, we need a better way for CLI to understand which context it is operating now. Using `samdev` for now is OK but I will add a TODO here to investigate it further during release improvements in the future.